File: code/pat/b1001.ts
// code/pat/b1001.ts 害死人不偿命的(3n+1)猜想 https://pintia.cn/problem-sets/994805260223102976/problems/994805325918486528
/**
 *
 * @param {number} num
 * @returns {number}
 */
export function solution(num: number): number {
  let count = 0;
  while (num !== 1) {
    if (num % 2 === 0) {
      num = num >> 1;
    } else {
      num = (3 * num + 1) >> 1;
    }
    count++;
  }
  return count;
}

File: code/pat/java/B1001.java
package code.pat.java;

import java.util.Scanner;

public class B1001 {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    while (in.hasNextInt()) {
      int num = in.nextInt();
      int count = 0;
      while (num != 1) {
        if (num % 2 == 0) {
          num = num >> 1;
        } else {
          num = (3 * num + 1) >> 1;
        }
        count++;
      }
      System.out.println(count);
    }
  }
}